packages/remix-dev/compiler.ts

```diff
@@ -427,7 +427,12 @@ async function createServerBuild(
           config.serverBuildTarget
         ),
       mainFields:
-        config.serverModuleFormat === "esm"
+        !!config.serverBuildTarget &&
+        ["cloudflare-workers", "cloudflare-pages"].includes(
+          config.serverBuildTarget
+        )
+          ? ["browser", "module", "main"]
+          : config.serverModuleFormat === "esm"
           ? ["module", "main"]
```

Comment:
Maybe the conditional could be encapsulated into a descriptive variable name to give history & context for the to the logic

For example:
```ts
const isCloudFlareWorkerRuntime =  ["cloudflare-workers", "cloudflarepages"].includes(config?.serverBuildTarget)


mainFields: isCloudFlareWorkerRuntime  ? ...
```

Reply:
I think you're right and I'll fix it.

Reply:
Fixed